share/completions/magento.fish

```diff
@@ -0,0 +1,720 @@
+    for i in $commands
+        echo $i
+    end
+end
+
+#########################################################
+
+function __fish_magento_not_in_command -d "Checks that prompt is not inside of magento command"
+    for i in (commandline -opc)
+        if contains -- $i (__fish_print_magento_commands_list)
+            return 1
+        end
+    end
+    return 0
+end
+
+#########################################################
+
+# Perforce command is a single word that comes either as first argument
+# or directly after global options.
+# To test whether we're in command, it's enough that a command will appear
+# in the arguments, even though if more than a single command is specified,
+# p4 will complain.
+function __fish_magento_is_using_command -d "Checks if prompt is in a specific command"
+    if contains -- $argv[1] (commandline -opc)
+        return 0
+    end
+    return 1
+end
+
+#########################################################
+
+function __fish_magento_register_command -d "Adds a completion for a specific command"
+    complete -c magento -n "__fish_magento_not_in_command" -a $argv[1] $argv[2..-1]
+end
+
+#########################################################
+
+function __fish_magento_register_command_option -d "Adds a specific option for a command"
+    complete -c magento -n "__fish_magento_is_using_command $argv[1]" $argv[2..-1]
+end
+
+
+
+##################
+# Global options #
+##################
+complete -x -c magento -s h -l help -d "Displays help for a command";
+complete -x -c magento -s q -l quiet -d "Do not output any message";
+complete -x -c magento -s v -l verbose -a "(__fish_print_magento_verbosity_levels)" -d "Increase the verbosity of messages: 1 for normal output, 2 for more verbose output and 3 for debug";
+complete -x -c magento -o -vv;
+complete -x -c magento -o vvv;
+complete -x -c magento -s V -l version        -d "Display this application version";
+complete -x -c magento      -l ansi           -d "Force ANSI output";
+complete -x -c magento      -l no-ansi        -d "Disable ANSI output";
+complete -x -c magento -s n -l no-interaction -d "Do not ask any interactive question";
+
+################
+# Sub-Commands #
+################
+__fish_magento_register_command help -a "(__fish_print_magento_commands_list)" -d "Displays help for a command";
+__fish_magento_register_command list -d "Lists commands";
+
+__fish_magento_register_command admin:user:create -x -d "Creates an administrator";
+__fish_magento_register_command admin:user:unlock -x -d "Unlock Admin Account";
+
+__fish_magento_register_command app:config:dump -d "Create dump of application";
+
+__fish_magento_register_command cache:clean -d "Cleans cache type(s)";
+__fish_magento_register_command cache:disable -d "Disables cache type(s)";
```

Comment:
Just use the plural - "Disable cache types". I'm assuming this means it can take multiple arguments?